doc/source/train/examples/pytorch/pytorch-profiling/README.md

```diff
@@ -0,0 +1,465 @@
+                
+                prof.step()
+                
+                # Log performance metrics every 50 batches
+                if batch_idx % 50 == 0 and world_rank == 0:
+                    print(f"Epoch {epoch}, Batch {batch_idx}, Loss: {loss.item():.4f}")
+
+            epoch_end_time.record()
+            torch.cuda.synchronize()
+            
+            # [4] Calculate and report timing metrics.
+            # ======================================
+            epoch_time = epoch_start_time.elapsed_time(epoch_end_time)
+            metrics = {
+                "loss": loss.item(), 
+                "epoch": epoch,
+                "epoch_time_ms": epoch_time,
+                "profiler_step": prof.step_num
+            }
+            
+            with tempfile.TemporaryDirectory() as temp_checkpoint_dir:
+                torch.save(
+                    model.state_dict(),
+                    os.path.join(temp_checkpoint_dir, "model.pt")
+                )
+                ray.train.report(
+                    metrics,
+                    checkpoint=ray.train.Checkpoint.from_directory(temp_checkpoint_dir),
+                )
+            
+            if world_rank == 0:
+                print(f"Epoch {epoch} completed in {epoch_time:.2f}ms")
+
+    # [5] Export comprehensive profiling data.
+    # ======================================
+    run_name = ray.train.get_context().get_experiment_name()
+    
+    # Export memory timeline
+    prof.export_memory_timeline(
+        f"/mnt/cluster_storage/{run_name}/rank{world_rank}_advanced_memory_profile.html"
+    )
+    
+    
+    if world_rank == 0:
+        print(f"Advanced profiling complete! Check '/mnt/cluster_storage/{run_name}/' for detailed profiling results.")
+        print("Files generated:")
+        print(f"  - rank{world_rank}_advanced_memory_profile.html (Memory analysis)")
+        print(f"  - rank{world_rank}_chrome_trace.json (Chrome trace)")
+        print("  - TensorBoard logs in /mnt/cluster_storage/logs/advanced/")
+
+```
+
+Kick off the Ray Train job similarly to the previous step.
+
+
+```python
+# Run the advanced profiling example
+scaling_config = ray.train.ScalingConfig(num_workers=1, use_gpu=True)
+
+# Create a unique experiment name for advanced profiling
+advanced_experiment_name = f"advanced_profiling_{uuid.uuid4().hex[:8]}"
+
+# Configure run settings
+run_config = ray.train.RunConfig(
+    storage_path="/mnt/cluster_storage/",
+    name=advanced_experiment_name,
+)
+
+# Launch advanced profiling training job
+trainer = ray.train.torch.TorchTrainer(
+    train_func_advanced_profiling,
+    scaling_config=scaling_config,
+    run_config=run_config,
+)
+
+print(f"Starting advanced profiling training: {advanced_experiment_name}")
+result = trainer.fit()
+print("Advanced profiling training completed successfully!")
+print(f"Check '/mnt/cluster_storage/{advanced_experiment_name}/' for comprehensive profiling results.")
```

Comment:
nit: maybe also print `result` so users can see the profiling metrics that they reported